pkg/services/hcloud/remediation/remediation.go

```diff
@@ -49,6 +50,9 @@ func NewService(scope *scope.HCloudRemediationScope) *Service {
 // Reconcile implements reconcilement of HCloudRemediation.
 func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err error) {
 	server, err := s.findServer(ctx)
+	if errors.Is(err, hcloudutil.ErrNilProviderID) {
```

Comment:
shouldn't we stop the remediation in this case rather? Like we do below with "phase deleting"?